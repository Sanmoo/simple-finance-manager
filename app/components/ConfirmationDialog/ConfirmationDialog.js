/*
 *
 * ConfirmationDialog
 *
 */

import React, { memo } from 'react';
import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogContentText from '@material-ui/core/DialogContentText';
import DialogTitle from '@material-ui/core/DialogTitle';
import PropTypes from 'prop-types';

export const ConfirmationDialog = ({
  opened,
  onDismiss,
  onConfirm,
  title,
  content,
  onConfirmText,
  onDismissText,
}) => (
  <div>
    <Dialog
      open={opened}
      onClose={onDismiss}
      aria-labelledby="alert-dialog-title"
      aria-describedby="alert-dialog-description"
    >
      <DialogTitle id="alert-dialog-title">{title}</DialogTitle>
      <DialogContent>
        <DialogContentText id="alert-dialog-description">
          {content}
        </DialogContentText>
      </DialogContent>
      <DialogActions>
        <Button onClick={onDismiss} color="primary">
          {onDismissText}
        </Button>
        <Button onClick={onConfirm} color="primary" autoFocus>
          {onConfirmText}
        </Button>
      </DialogActions>
    </Dialog>
  </div>
);

ConfirmationDialog.propTypes = {
  opened: PropTypes.bool.isRequired,
  onDismiss: PropTypes.func.isRequired,
  title: PropTypes.string,
  onConfirm: PropTypes.func.isRequired,
  onConfirmText: PropTypes.string,
  content: PropTypes.string.isRequired,
  onDismissText: PropTypes.string,
};

ConfirmationDialog.defaultProps = {
  title: 'Atenção',
  onConfirmText: 'Confirm',
  onDismissText: 'Cancel',
};

export default memo(ConfirmationDialog);
