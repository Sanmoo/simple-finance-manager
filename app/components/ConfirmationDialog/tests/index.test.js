/**
 *
 * Tests for ConfirmationDialog
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import ConfirmationDialog from '../ConfirmationDialog';

describe('<ConfirmationDialog />', () => {
  let props;

  beforeEach(() => {
    props = {
      opened: false,
      onDismiss: jest.fn(),
      title: 'My Title',
      onConfirm: jest.fn(),
      onConfirmText: 'Confirm',
      content: 'My content',
      onDismissText: 'Dismiss',
    };
  });

  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<ConfirmationDialog {...props} />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<ConfirmationDialog {...props} />);
    expect(firstChild).toMatchSnapshot();
  });
});
