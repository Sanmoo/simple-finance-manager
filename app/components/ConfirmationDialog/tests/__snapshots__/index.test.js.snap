// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationDialog /> Should render and match the snapshot 1`] = `<div />`;
